### ISSUE TITLE
derby.timer.jar - eTekGadget -No Results if gate is held open for less than 0.5s.

### ISSUE THREAD

**Issue by JonG98** [jeffpiazza/derbynet#204]
derby.timer.jar not providing results in Linux or Windows10.  Issued was corrected when I downloaded the derby.timer.jar version that is attached in Issue#180.

Information - 
Timer Version Returned from Device:
"The Champ" (SmartLine/BestTrack)
eTekGadget SmartLine Timer v20.09 (B0010)

Here are the results from Windows 10 Timer Running with Local server.

[Goodtimer.log](https://github.com/jeffpiazza/derbynet/files/8312062/Goodtimer.log)

[BadTimerLog.txt](https://github.com/jeffpiazza/derbynet/files/8312065/BadTimerLog.txt)

**Comment by jeffpiazza** [comment]
Could you export a copy of the server's database (from the About page) and attach here?

The "goodtimer.log" shows evidence of the #200 bug (e.g., race timing out at 15:48:27.206), while the "BadTimerLog.txt" shows good results being posted to the server (e.g., message=FINISHED&roundid=1&heat=1&lane1=2.9108&lane2=2.5571&lane3=2.0852&lane4=1.7214).  Both seem to be truncated at about 31KB, although GitHub docs claim attached files can be up to 25MB or so.  

Could describe more specifically what problem you're seeing?  Would it be possible to attach the complete timer logs?

**Comment by JonG98** [comment]
Jeff,

I think I can recreate the issue.  It appears that if the timer does not have gate left open the whole time.  The results do not update.  I wonder if my start guy was closing the gate or our start switch has an issue where it gets hit but does not stay made for the whole race.

[CleanPackDownload.xml.txt](https://github.com/jeffpiazza/derbynet/files/8312483/CleanPackDownload.xml.txt)

[PiTimerLogFIles.zip](https://github.com/jeffpiazza/derbynet/files/8312484/PiTimerLogFIles.zip)

**Comment by JonG98** [comment]
I ran a little test with PuTTY to see what results I could get.  

I am almost 95% confident if you issue an rs command after the rg command the rs command returns 0, you have to reissue the rg command.  As long as the rs command returns 1 you do not have to reissue the rg command.

If you do issue the rg command, it will return the results. when the heat is finished.

Then I am thinking since you never send the r command since you never got results, the system just stays hung from your timer's perspective.

Good
[GateStaysOpenTillEnd.txt](https://github.com/jeffpiazza/derbynet/files/8312646/GateStaysOpenTillEnd.txt)

Bad
[GateClosedBeforeEnd.txt](https://github.com/jeffpiazza/derbynet/files/8312647/GateClosedBeforeEnd.txt)

Issue 200 - This does not issue the RS Command and waits to get the results and hence it works even if the gate is closed before the end of the heat.
[timer-20220320-2147.log](https://github.com/jeffpiazza/derbynet/files/8312648/timer-20220320-2147.log)

**Comment by jeffpiazza** [comment]
The "bad" log above (https://github.com/jeffpiazza/derbynet/issues/204#issuecomment-1073411615) shows the gate being opened at +21:42:15.370 through +21:42:15.745, and then being reported closed again at +21:42:15.933.

There's a --min-gate-time flag with a default 500ms value; the starting gate needs to remain open at least that long in order for the host side to have confidence that the gate has actually been opened.  That is, there needs to be a "gate (still) open" report received at least 500ms after the first "gate open" result.  Closing the gate before that much time has elapsed will suppress sending the "rg" command to ask for heat results, and will instead continue polling the gate state.  

You could experiment with a shorter min value if you think it's more suitable, possibly even setting the flag to 0.  The 500ms default value may be overly long, although I've not personally seen a track set-up where an operator could close the gate is less than a half second after starting the race, even if they intended to.  Perhaps your installation is different, though.

**Comment by JonG98** [comment]
I did test it again with making sure that I held the gate open till the timer showed : "INT SET >--RACE_STARTED--> RUNNING:.  I then released it and flagged the lanes.  It all appears to work correctly on my test bench.  

I also did one bad one (<0.1s it missed that timing), but I reset and did it correctly and it got that timing correctly.

I think we have a bad switch or the gate is not making full contact with the switch.  I will have to look at the mechanicals at some point.  Thank you so much for your help.
